### PR TITLE
updates ++boil:ap to crash if a book has pages with the same tag

### LINKED ISSUE #296: $% doesn't error on duplicate stems
This shouldn't compile:

```
> $%({$a @} {$a @ @})
<1.ziw {* {him/@p our/@p now/@da eny/@uvJ} $~ <288.euk 42.dit 409.spy 110.xht 1.ztu $151>}>
```

Because the purpose of `$%` "kelp" is to discriminate orthogonal cases, and result is not a proper kelp:

```
> ($%({$a @} {$a @ @}) %a 1 1)
[%a 0]
```

HT @Fang- for demonstrating this in ~talkbot

### PR DESCRIPTION
**Pull request by joemfb**
I've only tested this in a generator -- I can't find a recent commit of master that lets me `|reset` ...

closes #296

### CODE REVIEW

**Comment by joemfb** [comment]
Tested on a maintenance fake-zod. This should be good.

**Comment by juped** [comment]
Just metal and boot fresh to test, for the record

**Comment by joemfb** [comment]
Something else is must be wrong, i was able to reset with #364.

If I copy this change into the pier and restart, a reset is attempted, but it fails with several long stack traces. Problem is the paths aren't printing. I wonder if there are duplicate pages in the vanes on master ...

**Comment by joemfb** [comment]
There was a duplicate `$bunt` page ...

There's also something wrong with printf hints -- I tried many different combinations before I found one that would actually print the duplicate page head.

This works if I copy the commits one at a time, restarting for two separate resets. If I do them all at once, I get this:

```
[%hoon-compile ~polhet-tantug]
hoon-assembly
[%compile-arvo ~nismer-firner ~lagluc-podtyv]
arvo-boot
%hoon-load
[%tang /~zod/home/~2017.3.5..06.28.03..e882/sys/zuse ~socber-fonfed]
[%vane %a /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/ames ~bannul-hanwyn]
%ames-reload
[%vane %b /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/behn ~hinnel-milmyr]
[%vane %c /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/clay ~docbyt-datmyn]
book-dup-page: bunt
-find.$.+2
/:<[66 17].[66 73]>
/:<[66 5].[66 74]>
%load-lost
/:<[65 5].[66 74]>
/:<[64 5].[66 74]>
/:<[63 5].[66 74]>
/:<[62 5].[66 74]>
/:<[61 5].[66 74]>
[%vane %d /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/dill ~labsem-pasrym]
[%vane %e /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/eyre ~risdyn-ralfus]
book-dup-page: bunt
-find.$.+2
/:<[66 17].[66 73]>
/:<[66 5].[66 74]>
%load-lost
/:<[65 5].[66 74]>
/:<[64 5].[66 74]>
/:<[63 5].[66 74]>
/:<[62 5].[66 74]>
/:<[61 5].[66 74]>
[%vane %f /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/ford ~nocsep-salbyt]
[%vane %g /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/gall ~fipnec-hinbel]
book-dup-page: bunt
-find.$.+2
/:<[66 17].[66 73]>
/:<[66 5].[66 74]>
%load-lost
/:<[65 5].[66 74]>
/:<[64 5].[66 74]>
/:<[63 5].[66 74]>
/:<[62 5].[66 74]>
/:<[61 5].[66 74]>
[%vane %j /~zod/home/~2017.3.5..06.28.03..e882/sys/vane/jael ~ponnub-linmex]
: /~zod/home/4/sys/hoon/hoon
<<<reset>>>
```

... which seems to suggest that different hoon versions are being used for different vanes (?!?)

**Comment by cgyarvin** [comment]
That's not possible and this is extremely bizarre.  Well, okay, maybe it's
possible.  Master is in a strange state with half old and half new kernel
control...

File: sys/hoon.hoon [hunk 1]
```diff
@@ -5426,7 +5426,6 @@
     {$conl p/(list twig)}                                ::  :~ [p ~]
     {$conp p/(list twig)}                                ::  :* p as a tuple
   ::                                            ::::::  invocations
-    {$bunt p/twig}                                      ::  %$
```
Comment (juped) [inline sys/hoon.hoon]: ?

Comment (joemfb) [inline sys/hoon.hoon]: it's a duplicate page

Comment (juped) [inline sys/hoon.hoon]: I see

**Comment by joemfb** [comment]
I've cherry-picked this onto master, and updated the now-working printf (referenced above).

Note: this does not `|reset` cleanly, do to a duplicate `%g` tag in `++sign:able:xmas`.

**Comment by belisarius222** [comment]
@joemfb  looks like this needs another compatibility bump to merge into master.

**Comment by joemfb** [comment]
Rebased forward. Note that there is still a duplicate `%g` tag in %zuse preventing a clean `|reset`.

**Comment by ohAitch** [comment]
Ha, well caught. It's in ++xmas which is not currently live; probably just
change it to $? in the interim?

**Comment by belisarius222** [comment]
Looks good to me.

**Comment by joemfb** [comment]
@belisarius222, had you seen 87e658a765549e9674377d9beeafc3dc6529322b?

**Comment by belisarius222** [comment]
I had not, and still don't know exactly what it means. Does its failure imply that I shouldn't have merged this?

**Comment by belisarius222** [comment]
Oh, looks like it reverts this PR ... I guess I should revert it again, considering it apparently broke something.

**Comment by joemfb** [comment]
@cgyarvin, do you remember what the issue was with these changes?

**Pull request merged**